### PR TITLE
refactor!: breaking api changes

### PR DESCRIPTION
**Pull request by agazso**
This PR contains fixes for the breaking changes for the upcoming 0.5.0 Bee release.
The main breaking changes:
- the `/tags` endpoint returns different values
- uploading a chunk to the `/chunks` endpoint now returns the reference
- there is a new `/soc` endpoint to upload single owner chunks

### CODE REVIEW

**Comment by vojtechsimetka** [comment]
One suggestion. If this is breaking change on our interface, you may want to add `!` after the scope (in this case `refactor!: ...` ). This way it will be properly caught by the `please-release` system and it will be listed in the breaking changes.

File: src/chunk/upload.ts [hunk 1]
```diff
@@ -12,9 +13,15 @@ import * as chunkAPI from '../modules/chunk'
  * @param chunk     A chunk object
  * @param options   Upload options
  */
-export function uploadChunk(url: string, chunk: Chunk, options?: UploadOptions): Promise<BeeResponse> {
-  const address = chunk.address()
-  const hash = bytesToHex(address)
+export function uploadSingleOwnerChunk(
```
Comment (nugaon) [inline src/chunk/upload.ts:16]: It is purely only a SOC function, that could be moved to `src/chunk/soc.ts`

**Review by nugaon — CHANGES_REQUESTED** [review]

File: src/chunk/soc.ts [hunk 2]
```diff
@@ -67,22 +51,36 @@ export function isValidSingleOwnerChunkData(
...
+    const identifier = verifyBytesAtOffset(SOC_IDENTIFIER_OFFSET, IDENTIFIER_SIZE, data)
+    const socAddress = keccak256Hash(identifier, ownerAddress)
+
+    if (!bytesEqual(address, socAddress)) {
+      throw new BeeError('verifySingleOwnerChunk')
+    }
+
+    return makeSingleOwnerChunkFromData(data, address, ownerAddress)
+  } catch (e) {
+    throw e
```
[the remark this answers is not in the document]
Comment (agazso) [reply src/chunk/soc.ts:65]: Good catch, I will remove the `try ... catch`

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -34,7 +34,8 @@ jobs:
           export URL=$(curl -s https://api.github.com/repos/ethersphere/bee-local/releases/latest | jq -r .tarball_url)
           curl -Ls ${URL} -o bee-local.tar.gz
           tar --strip-components=1 --wildcards -xzf bee-local.tar.gz ethersphere-bee-local-*/{beeinfra.sh,helm-values,hack}
-          sed -i 's/IMAGE_TAG="latest"/IMAGE_TAG="0.4.2"/' beeinfra.sh
+          # TODO remove comment and update version number once the release is out
+          # sed -i 's/IMAGE_TAG="latest"/IMAGE_TAG="0.4.2"/' beeinfra.sh
```
Comment (vojtechsimetka) [inline .github/workflows/ci.yaml:38]: Just so we don't forget about this :)
```suggestion
          sed -i 's/IMAGE_TAG="latest"/IMAGE_TAG="0.5.0"/' beeinfra.sh
```

**Review by vojtechsimetka — APPROVED** [review]

File: src/modules/debug/settlements.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ const settlementsEndpoint = '/settlements'
 
 export interface Settlements {
   peer: string
-  received: number
+  received: number | null
```
Comment (nugaon) [inline src/modules/debug/settlements.ts:7]: I would write `received?: number` instead of that

File: test/modules/debug/settlements.spec.ts [hunk 5]
```diff
@@ -25,7 +25,9 @@
       const peerSettlementResponse = await settlements.getSettlements(beeDebugUrl(), peerSettlement.peer)
 
       expect(peerSettlementResponse.peer).toEqual(peerSettlement.peer)
-      expect(typeof peerSettlementResponse.received).toBe('number')
+      // FIXME this is due to a bug in Bee
```
Comment (nugaon) [inline test/modules/debug/settlements.spec.ts:28]: you commented 12 hours ago it is fixed in Bee PR [#1213](https://github.com/ethersphere/bee/pull/1213).
could you correct this?

**Review by nugaon — APPROVED** [review]

**Pull request merged**